_generated/AzureTestPlanV0/testLibExecutor.ts

```diff
@@ -0,0 +1,88 @@
+import os = require('os');
+import Q = require('q');
+import path = require('path');
+import fs = require('fs');
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as tr from 'azure-pipelines-task-lib/toolrunner';
+
+var isWindows = os.type().match(/^Win/);
+var m2HomeEnvVar: string = null;
+var mavenPOMFile: string = tl.getPathInput('mavenPOMFile', true, true);
+
+function getMavenExec(){
+    var mvnExec: string = '';
+    m2HomeEnvVar = tl.getVariable('M2_HOME');
+    if (m2HomeEnvVar) {
+        tl.debug('Using M2_HOME environment variable value for Maven path: ' + m2HomeEnvVar);
+        mvnExec = path.join(m2HomeEnvVar, 'bin', 'mvn');
+    }
+    // Second, look for Maven in the system path
+    else {
+        tl.debug('M2_HOME environment variable is not set, so Maven will be sought in the system path');
+        mvnExec = tl.which('mvn', true);
+    }
+
+    if (isWindows &&
+        !mvnExec.toLowerCase().endsWith('.cmd') &&
+        !mvnExec.toLowerCase().endsWith('.bat')) {
+        if (tl.exist(mvnExec + '.cmd')) {
+            // Maven 3 uses mvn.cmd
+            mvnExec += '.cmd';
+        }
+        else if (tl.exist(mvnExec + '.bat')) {
+            // Maven 2 uses mvn.bat
+            mvnExec += '.bat';
+        }
+    }
+
+    tl.checkPath(mvnExec, 'maven path');
+    tl.debug('Maven executable: ' + mvnExec);
+
+    return mvnExec;
+}
+
+function getExecOptions(): tr.IExecOptions {
```

Comment:
nit suggestion - consider building an interface of testlibexecutor/invokers and establish a composition relationship with each frameworks like maven, pytest etc similar to ITestResultParser and TrxResultsparser and NunitResultsParser etc,  This would help while onboarding new languages/frameworks in future.

Reply:
Sure. Once, we have the task working for P0 scenarios covered hosted and selfhosted. We will take up the refactoring of the code.